packages/core-transaction-pool-mem/lib/storage.js

```diff
@@ -0,0 +1,100 @@
+'use strict'
+
+const logger = require('@arkecosystem/core-container').resolvePlugin('logger')
+
+/**
+ * A permanent storage (on-disk), supporting some basic functionalities required
+ * by the transaction pool.
+ */
+class Storage {
+  /**
+   * Construct the storage.
+   */
+  constructor () {
+    this.file = 'transaction_pool.sqlite'
+    this.table = 'pool'
+
+    const BetterSqlite3 = require('better-sqlite3')
+    this.db = new BetterSqlite3(this.file)
+
+    const newMode = this.db.pragma('journal_mode=WAL', true)
+    if (newMode.toUpperCase() !== 'WAL') {
+      logger.warn(
+        'Transaction pool: could not switch SQLite journal mode to WAL. ' +
+        `The journal mode is '${newMode}'. Expect reduced performance.`)
+    }
+
+    this.db.exec(`
+      CREATE TABLE IF NOT EXISTS ${this.table} (
+        "sequence" INTEGER PRIMARY KEY AUTOINCREMENT,
+        "id" VARCHAR(64) UNIQUE,
+        "senderPublicKey" VARCHAR(66) NOT NULL,
+        "serialized" TEXT NOT NULL
```

Comment:
I would replace this with `"serialized" bytea NOT NULL` to be consistent with how the `transactions` table stores the serialized hex.

Reply:
https://www.postgresql.org/docs/10/static/datatype-binary.html#id-1.5.7.12.9

https://www.sqlite.org/datatype3.html

no `bytea` in SQLite?

Reply:
I edited my comment shortly after, `BLOB` should be the equivalent in SQLite.

Reply:
To illustrate:
```
from db (binary) ->
convert to hex [1] ->
pass to the constructor of Transaction ->
pass to Transaction.deserialize() ->
convert to binary using ByteBuffer.fromHex(hexString, true) [2] ->
play with the binary
```
[1] and [2] can/should be removed. But this is outside of the scope of this tx mem pool / redis replacement.

Reply:
Yes, I agree. Let's reiterate that once mainnet is on v2.